scripts/gh-analytics.js

```diff
@@ -0,0 +1,55 @@
+const { Octokit } = require("@octokit/core");
+const Analytics = require('analytics-node');
+const uniqid = require('uniqid'); 
+
+async function retrieveAnalytics () {
+  const octokit = new Octokit({
+    auth: process.env.GH_TOKEN
+  })
+  const analytics = new Analytics(process.env.SEGMENT_API_KEY);
+  const date = new Date();
+  
+  console.log("Retrieving referrals...")
+  const referrersResponse = await octokit.request('GET /repos/{owner}/{repo}/traffic/popular/referrers', {
+    owner: 'medusajs',
+    repo: 'medusa'
+  })
+  console.log(`Pushing ${referrersResponse.data.length} referrers...`);
+  referrersResponse.data.forEach((ref) => {
+    if (!process.env.DEBUG) {
+      analytics.page({
+        anonymousId: uniqid(),
+        category: 'referrals',
+        properties: {
+          title: ref.referrer,
+          count: ref.uniques
+        },
+        timestamp: date
+      })
+    }
+  })
+  
+  console.log("Retrieving popular paths...")
+  const pathsResponse = await octokit.request('GET /repos/{owner}/{repo}/traffic/popular/paths', {
+    owner: 'medusajs',
+    repo: 'medusa'
+  })
+  console.log(`Pushing ${pathsResponse.data.length} paths...`);
+  
+  pathsResponse.data.forEach((path) => {
+    if (!process.env.DEBUG) {
+      analytics.page({
```

Comment:
would suggest using a more generic track here as the typical segment page event is meant to semantically signify a "real" visit of a page which is not really he case here.

Reply:
This one is actually more like the page event in Segment compared to the `referral` one. Looking at segment's events, it expects `path`, `title`, and `url`. I can generate the URL here but I'm not sure for referrals what we can do since the API does not product that kind of data.

Reply:
The thing is just that we are not in reality recording a single page view here as Segment's Spec expects, so it becomes a bit shoehorning. Think it would be better to have something like this:

      analytics.track({
        event: "popular_paths",
        properties: {
          title: path.title,
          path: path.path,
          count: path.uniques
        },
        timestamp: date
      })

Would apply the same idea to the referrer paths.

Reply:
Oh ok so instead of a page we change it to a track. Makes sense 🙂

There's another issue here but I'm not sure whether we should look into resolving it here or in Metabase: This API allows us to retrieve the popular paths and referrals in the past 14 days. Would we be able to retrieve the data bi-weekly on Metabase using this data the way it's collected here?